tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -539,15 +539,23 @@ class TestInnerFileStruct(object):
         a: JPEGImageFile
         b: typing.List[FlyteFile]
         c: typing.Dict[str, FlyteFile]
+        d: typing.List[FlyteFile]
+        e: typing.Dict[str, FlyteFile]
 
     @dataclass_json
     @dataclass
     class TestFileStruct(object):
         a: FlyteFile
         b: TestInnerFileStruct
 
-    f = FlyteFile("s3://tmp/file")
-    o = TestFileStruct(a=f, b=TestInnerFileStruct(a=JPEGImageFile("s3://tmp/file.jpeg"), b=[f], c={"hello": f}))
+    remote_path = "s3://tmp/file"
+    f1 = FlyteFile(remote_path)
+    f2 = FlyteFile("/tmp/file")
```

Comment:
Can you also add a few tests for the other supported types (FlyteDirectory, StructuredDataset, etc)?